flexmeasures/api/v3_0/assets.py

```diff
@@ -0,0 +1,230 @@
+        """
+        asset = GenericAsset(**asset_data)
+        db.session.add(asset)
+        db.session.commit()
+        return asset_schema.dump(asset), 201
+
+    @route("/<id>", methods=["GET"])
+    @use_kwargs({"asset": AssetIdField(data_key="id")}, location="path")
+    @permission_required_for_context("read", arg_name="asset")
+    @as_json
+    def fetch_one(self, id, asset):
+        """Fetch a given asset.
+
+        .. :quickref: Asset; Get an asset
+
+        This endpoint gets an asset.
+
+        **Example response**
+
+        .. sourcecode:: json
+
+            {
+                "generic_asset_type_id": 2,
+                "name": "Test battery",
+                "id": 1,
+                "latitude": 10,
+                "longitude": 100,
+                "account_id": 1,
+            }
+
+        :reqheader Authorization: The authentication token
+        :reqheader Content-Type: application/json
+        :resheader Content-Type: application/json
+        :status 200: PROCESSED
+        :status 400: INVALID_REQUEST, REQUIRED_INFO_MISSING, UNEXPECTED_PARAMS
+        :status 401: UNAUTHORIZED
+        :status 403: INVALID_SENDER
+        :status 422: UNPROCESSABLE_ENTITY
+        """
+        return asset_schema.dump(asset), 200
+
+    @route("/<id>", methods=["PATCH"])
+    @use_args(AssetSchema(partial=True))
+    @use_kwargs({"db_asset": AssetIdField(data_key="id")}, location="path")
+    @permission_required_for_context("update", arg_name="db_asset")
+    @as_json
+    def patch(self, asset_data: dict, id: int, db_asset: GenericAsset):
+        """Update an asset given its identifier.
+
+        .. :quickref: Asset; Update an asset
+
+        This endpoint sets data for an existing asset.
+        Any subset of asset fields can be sent.
+
+        Several fields are not allowed to be updated, e.g. id. They are ignored.
+
+        **Example request**
+
+        .. sourcecode:: json
+
+            {
+                "latitude": 11.1,
+                "longitude": 99.9,
+            }
+
+
+        **Example response**
+
+        The whole asset is returned in the response:
+
+        .. sourcecode:: json
+
+            {
+                "generic_asset_type_id": 2,
+                "id": 1,
+                "latitude": 11.1,
+                "longitude": 99.9,
+                "name": "Test battery",
+                "account_id": 2,
+            }
+
+        :reqheader Authorization: The authentication token
+        :reqheader Content-Type: application/json
+        :resheader Content-Type: application/json
+        :status 200: UPDATED
+        :status 400: INVALID_REQUEST, REQUIRED_INFO_MISSING, UNEXPECTED_PARAMS
+        :status 401: UNAUTHORIZED
+        :status 403: INVALID_SENDER
+        :status 422: UNPROCESSABLE_ENTITY
+        """
+        ignored_fields = ["id", "account_id"]
+        for k, v in [(k, v) for k, v in asset_data.items() if k not in ignored_fields]:
+            setattr(db_asset, k, v)
```

Comment:
I believe it's not uneditable out of principle, but because it was a hassle to implement while moving to an internal asset API. I'd leave it editable, and probably the UI should some day in the future allow that, as well. Of course we might study it and find that at some point there would be weird consequences, but for now I leave that aside.